mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list.go

```diff
@@ -0,0 +1,415 @@
+		return
+	}
+
+	entry, ok := accessList.(*matlas.ProjectIPAccessList)
+	if !ok {
+		resp.Diagnostics.AddError("error", errorAccessListCreate)
+		return
+	}
+
+	projectIPAccessListNewModel := newTFProjectIPAccessListModel(projectIPAccessListModel, entry)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &projectIPAccessListNewModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func newTFProjectIPAccessListModel(projectIPAccessListModel *tfProjectIPAccessListModel, projectIPAccessList *matlas.ProjectIPAccessList) *tfProjectIPAccessListModel {
+	entry := projectIPAccessList.IPAddress
+	if projectIPAccessList.CIDRBlock != "" {
+		entry = projectIPAccessList.CIDRBlock
+	} else if projectIPAccessList.AwsSecurityGroup != "" {
+		entry = projectIPAccessList.AwsSecurityGroup
+	}
+
+	id := fmt.Sprintf("%s-%s", projectIPAccessList.GroupID, entry)
+	return &tfProjectIPAccessListModel{
+		ID:               types.StringValue(id),
+		ProjectID:        types.StringValue(projectIPAccessList.GroupID),
+		CIDRBlock:        types.StringValue(projectIPAccessList.CIDRBlock),
+		IPAddress:        types.StringValue(projectIPAccessList.IPAddress),
+		AWSSecurityGroup: types.StringValue(projectIPAccessList.AwsSecurityGroup),
+		Comment:          types.StringValue(projectIPAccessList.Comment),
+		Timeouts:         projectIPAccessListModel.Timeouts,
+	}
+}
+
+func newMongoDBProjectIPAccessList(projectIPAccessListModel *tfProjectIPAccessListModel) []*matlas.ProjectIPAccessList {
+	return []*matlas.ProjectIPAccessList{
+		{
+			AwsSecurityGroup: projectIPAccessListModel.AWSSecurityGroup.ValueString(),
+			CIDRBlock:        projectIPAccessListModel.CIDRBlock.ValueString(),
+			IPAddress:        projectIPAccessListModel.IPAddress.ValueString(),
+			Comment:          projectIPAccessListModel.Comment.ValueString(),
+		},
+	}
+}
+
+func (r *ProjectIPAccessListRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var projectIPAccessListModelState *tfProjectIPAccessListModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &projectIPAccessListModelState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	entry := projectIPAccessListModelState.CIDRBlock.ValueString()
+	if projectIPAccessListModelState.IPAddress.ValueString() != "" {
+		entry = projectIPAccessListModelState.IPAddress.ValueString()
+	} else if projectIPAccessListModelState.AWSSecurityGroup.ValueString() != "" {
+		entry = projectIPAccessListModelState.AWSSecurityGroup.ValueString()
+	}
+
+	projectID := projectIPAccessListModelState.ProjectID.ValueString()
+	// Use the ID only with the IMPORT operation
+	var err error
+	if entry == "" && projectIPAccessListModelState.ID.ValueString() != "" {
+		projectID, entry, err = splitProjectIPAccessListImportID(projectIPAccessListModelState.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("the provided ID is not correct", err.Error())
+		}
+	}
```

Comment:
Can we move this logic over to the import function and replace the current `ImportStatePassthroughID` implementation?
From ImportState function documentation:

> ImportState is called when the provider must import the state of a resource instance. This method must return enough state so the Read method can properly refresh the full resource.

I see that in previous implementation we were processing the provided import id, defining the `project_id` attribute, as well as the encoded `id` so the Read operation would not have any import specific logic.

Reply:
Not in this case as we do not know what is the entry provided in the ID as it could be IP, CIDR or AWS security group. In order to move this logic in the import function, we should be able to know where in the state I should save the entry.

Reply:
Related question, but isn't the entry encoded in the resource id?
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_project_ip_access_list.go#L171-L174

If import function sets the proper encoded id, and project attribute
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_project_ip_access_list.go#L274-L281
I believe read operation can be defined with no import-specific logic.

Reply:
Updated based on our discussion in the 1:1. Thanks!